src/AspNet.Security.OAuth.SuperOffice/SuperOfficeAuthenticationHandler.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                throw new InvalidOperationException("An error occurred trying to obtain the context identifier from the current user's identity claims.");
+            }
+
+            // Add contextId to the Options.UserInformationEndpoint (https://sod.superoffice.com/{0}/api/v1/user/currentPrincipal).
+            string userInfoEndpoint = string.Format(CultureInfo.InvariantCulture, Options.UserInformationEndpoint, contextId);
+
+            // Get the SuperOffice user principal.
+            using var request = new HttpRequestMessage(HttpMethod.Get, userInfoEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+            using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException($"An error occurred when retrieving SuperOffice user information ({response.StatusCode}). Please check if the authentication information is correct and the corresponding SuperOffice API is enabled.");
+            }
+
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+            var context = new OAuthCreatingTicketContext(new ClaimsPrincipal(identity), properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+            context.RunClaimActions();
+            await Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        private async Task<string> ProcessIdTokenAndGetContactIdentifierAsync(
+            [NotNull] OAuthTokenResponse tokens,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] ClaimsIdentity identity)
+        {
+            string contextIdentifier = string.Empty;
+
+            string idToken = tokens.Response.RootElement.GetString("id_token");
+
+            if (Options.SaveTokens)
+            {
+                // Save id_token as well.
+                SaveIdToken(properties, idToken);
+            }
+
+            IEnumerable<Claim>? claims = null;
+
+            if (Options.ValidateTokens)
+            {
+                var tempClaimsPrincipal = await ValidateAsync(
+                    idToken,
+                    new TokenValidationParameters()
```

Comment:
Why are you creating a new `TVP` instance from scratch here? The instance coming from the options doesn't seem to be used, and it's super strange. You should be doing `Options.TokenValidationParameters.Clone()` instead.

Reply:
Ah, because I didn't see Clone. Thanks, I'll fix!